aiida_quantumespresso/common/workchain/base/restart.py

```diff
@@ -310,13 +267,24 @@ def _handle_calculation_failure(self, calculation):
 
         return
 
-    def _prepare_process_inputs(self, process, inputs):
-        """
-        Prepare the inputs dictionary for a calculation process. Any remaining bare dictionaries in the inputs
-        dictionary will be wrapped in a Dict data node except for the 'options' key which should remain
-        a standard dictionary. Another exception are dictionaries whose keys are not strings but for example tuples.
-        This is the format used by input groups as in for example the explicit pseudo dictionary where the key is
-        a tuple of kind to which the UpfData corresponds.
+    def _handle_unexpected_failure(self, calculation, exception=None):
+        """Handle an unexpected failure.
+
+        This occurs when a calculation excepted, was killed or finished with a non-zero exit status but no errors were
+        handled. If this is the second consecutive unexpected failure the work chain is aborted.
+
+        :param calculation: the calculation that failed in an unexpected way
+        :param exception: optional exception message to log to the report
+        :return: `ExitCode` if this is the second consecutive unexpected failure
         """
-        from aiida_quantumespresso.utils.mapping import prepare_process_inputs
-        return prepare_process_inputs(process, inputs)
+        if exception:
+            self.report('{}'.format(exception))
+
+        if self.ctx.unexpected_failure:
+            self.report('failure of {}<{}> could not be handled for the second consecutive time'.format(
+                self.ctx.calc_name, calculation.pk))
+            return self.exit_codes.ERROR_SECOND_CONSECUTIVE_UNHANDLED_FAILURE
+
+        self.ctx.unexpected_failure = True
+        self.report('failure of {}<{}> could not be handled, restarting once more'.format(
+            self.ctx.calc_name, calculation.pk))
```

Comment:
I would keep the "else:" clause and the indentation; it makes the code more readable IMO.

Reply:
This is actually against common style guides, which is also the reason that pylint will complain about such an unnecessary else-clause

Reply:
This guideline is not in PEP8, and its applicability [depends on context](https://dmerej.info/blog/post/else-after-return-yea-or-nay/). In this case it doesn't really matter (I can see your point of view: you could say that there are 2 precondition checks and a "main" code), but I would not let pylint's opinion about this override readability.